utils/build.config

```diff
@@ -6,7 +6,7 @@ ARGOBOTS = 9d48af08403bac649598942b5ee848c14600cd8a
 CART = 7bde2eaec684faa02372caca464b96136348aad4
 PMDK = 1.5.1
 ISAL = v2.26.0
-SPDK = 051297114cb393d3eb1169520d474e81b4215bf0
+SPDK = 0901744bb197f6d9fb6d8d120913dd7716bf8301
```

Comment:
Do we have to use an **unreleased** git hash here?  Is there not an official upstream release that will satisfy our needs?

Reply:
well, if we use some released version, then we have to hack our scons_local to copy out some SPDK internal headers for DAOS. I just submitted a patch to SPDK to export that API, and it's already being landed in this git hash.

Also, there is a build script defect in current SPDK, I suppose they'll provide a fix soon, then we can include that patch as well to simplify our building script for linking the SPDK libs.

Reply:
OK.  Sounds promising.  So to summarise, if we can use the most recent official release but then only need two patches to it to make it suitable for us, we should aim to do that for the RPM build at least.  That is usually done by backporting the patches to the release and adding them as `Patch*:` entries to the RPM specfile.

Non-RPM builds are a little bit less straightforward.  I do notice that they have a [v19.04.x maintenance branch](https://github.com/spdk/spdk/commits/v19.04.x) which we could push PRs for our desired patch backports.  That would at least provide an equivalent git commit to use in build.config.

Reply:
Yes. I've tried to upgrade to v19.04.1 in my local build, it requires only small changes to both DAOS and scons_local, but I'm still investigating on icc build failure problem.

Brian, I'm still not quite sure how should we cooperate the DAOS/scons_local updating with the updating in https://github.com/daos-stack/spdk ?  Also, could you help me on the spdk RPM spec updating? Thanks.

Reply:
So the spdk RPM update can live in a PR and be referenced from the PR by the various daos RPM build jobs until you are ready to land the scons_local and daos changes.  At that time you can land the spdk RPM changes.  It will cause a "flag day" event where everyone will need to update their work in progress since the spdk that their PRs will access will be the new one.

I can help with the RPM spec updating.  But before we go there, we should get all of the patches we want upstream on their  v19.04.x release branch.  Can you push some PRs for the changes you need to spdk to that branch in an effort to get them to land them?

Reply:
I see, thanks.

There is only one tiny patch needs be applied (and it's already in SPDK latest master), so can we just carry the patch and apply it when building RPM?

Reply:
So, in short, yes.  But if the RPM is going to be the upstream release 19.04.1 plus one patch, then `build.config` should reflect that and not be building 19.04.1 plus a whole bunch of patches (i.e. the tip -- or close to it) of master.

To get this equality -- so that RPMs are building what developers and testing is building -- something in spdk's git should be the same as what the RPM is building, so getting your patch landed to the  v19.04.x maintenance branch, and then pointing `build.config` at that, rather than the master commit which is many, many commits beyond 19.04.1 achieves that goal.

Reply:
Sure. I'll update build.config to use v19.04.1, also I'm going to update scons_local to apply the patch against v19.04.1.